xlive/Blam/Engine/game/game_results.cpp

```diff
@@ -52,9 +64,486 @@ c_game_results* game_results_get()
+{
+	//return INVOKE(0x66F43, 0, game_results_get_finalized_player_score, player_index);
+	int32 result = NONE;
+	c_game_results* game_results = game_results_get();
+
+	if (game_results->m_game_end_reason &&
+		game_results->m_initialized &&
+		game_results->m_players[player_index].exists)
+	{
+		result = game_results->m_players[player_index].score;
+	}
+
+	return result;
+}
+
+int32 game_results_get_finalized_player_place(int32 player_index)
+{
+	//return INVOKE(0x699BD, 0, game_results_get_finalized_player_place, player_index);
+	int32 result = NONE;
+	c_game_results* game_results = game_results_get();
+
+	if (game_results->m_game_end_reason &&
+		game_results->m_initialized &&
+		game_results->m_players[player_index].exists)
+	{
+		result = game_results->m_players[player_index].player_place;
+	}
+
+	return result;
+}
+
+s_player_configuration* game_results_get_finalized_player_configuration(int32 player_index)
+{
+	//return INVOKE(0x66FDC, 0, game_results_get_finalized_player_configuration, player_index);
+	s_player_configuration* result = nullptr;
+	c_game_results* game_results = game_results_get();
+
+	if (game_results->m_game_end_reason &&
+		game_results->m_initialized &&
+		game_results->m_players[player_index].exists)
+	{
+		result = &game_results->m_players[player_index].player_configuration;
+	}
+
+	return result;
+}
+
+int8* game_results_get_finalized_player_unknown_02(int32 player_index)
+{
+	//return INVOKE(0x67012, 0, game_results_get_finalized_player_unknown_02, player_index);
+	int8* result = nullptr;
+	c_game_results* game_results = game_results_get();
+
+	if (game_results->m_game_end_reason &&
+		game_results->m_initialized &&
+		game_results->m_players[player_index].exists)
+	{
+		result = game_results->m_players[player_index].unk_02;
+	}
+
+	return result;
+}
+
+e_game_team game_results_get_finalized_player_team(int32 player_index)
+{
+	//return INVOKE(0x67042, 0, game_results_get_finalized_player_team, player_index);
+	e_game_team result = _game_team_observer;
+	c_game_results* game_results = game_results_get();
+
+	if (game_results->m_game_end_reason &&
+		game_results->m_initialized &&
+		game_results->m_players[player_index].exists)
+	{
+		result = (e_game_team)game_results->m_players[player_index].player_configuration.team_index;
+	}
+
+	return result;
+}
+
+void game_results_get_finalized_player_profile_traits(int32 player_index, s_player_profile_traits* profile_traits)
+{
+	//INVOKE(0x6706D, 0, game_results_get_finalized_player_profile_traits, player_index, profile_traits);
+	c_game_results* game_results = game_results_get();
+
+	if (game_results->m_initialized && player_index != NONE && game_results->m_players[player_index].exists)
+	{
+		csmemcpy(
+			profile_traits, 
+			&game_results->m_players[player_index].player_configuration.profile_traits, 
+			sizeof(s_player_profile_traits));
+	}
+	else
+	{
+		csmemset(profile_traits, 0, sizeof(s_player_profile_traits));
```

Comment:
Proper function should be used here instead of csmemset

Reply:
I don't know if a memset is used here but should keep the original function that did the operation and do a memset or something else in there